functions/go/render-helm-chart/third_party/sigs.k8s.io/kustomize/api/builtins/HelmChartInflationGenerator.go

```diff
@@ -172,39 +187,16 @@ func (p *HelmChartInflationGeneratorPlugin) replaceValuesInline() error {
 	return err
 }
 
-// copyValuesFile to avoid branching.  TODO: get rid of this.
-func (p *HelmChartInflationGeneratorPlugin) copyValuesFile() (string, error) {
-	var b []byte
-	var err error
-	b, err = ioutil.ReadFile(p.ValuesFile)
-	if err != nil {
-		path := p.ValuesFile
-		if u, err := url.Parse(path); err == nil && (u.Scheme == "http" || u.Scheme == "https") {
-			var hc *http.Client
-			hc = &http.Client{}
-
-			resp, err := hc.Get(path)
-			if err != nil {
-				return "", err
-			}
-			defer resp.Body.Close()
-			b, err = ioutil.ReadAll(resp.Body)
-			if err != nil {
-				return "", err
-			}
-		}
-	}
-	return p.writeValuesBytes(b)
-}
-
 // Write a absolute path file in the tmp file system.
 func (p *HelmChartInflationGeneratorPlugin) writeValuesBytes(
-	b []byte) (string, error) {
+	b []byte, path string) (string, error) {
 	if err := p.establishTmpDir(); err != nil {
 		return "", fmt.Errorf("cannot create tmp dir to write helm values")
 	}
-	path := filepath.Join(p.tmpDir, p.Name+"-kustomize-values.yaml")
-	return path, ioutil.WriteFile(path, b, 0644)
+	// use a hash of the provided path to generate a unique, valid filename
+	hash := md5.Sum([]byte(path))
+	newPath := filepath.Join(p.tmpDir, p.Name+"-kustomize-values-"+hex.EncodeToString(hash[:])+".yaml")
```

Comment:
Is `hex.EncodeToString(md5.Sum(input))` a common practice in kustomize?

I guess `hex.EncodeToString(hash[:])` can be simplified to `hex.EncodeToString(hash)`, right?

Reply:
It is not something we do in kustomize, but the hasher we have in kustomize hashes an entire RNode rather than a string, the code is here: https://github.com/kubernetes-sigs/kustomize/blob/master/api/hasher/hasher.go

Alternatives here could be to use the kustomize's hasher and do something like `hasher.SortArrayAndComputeHash([]string{path})` - but we are not using this anywhere in kustomize so I am unsure how reliable it is. Or try something like the [kubernetes deep hasher object](https://github.com/kubernetes/kubernetes/blob/master/pkg/util/hash/hash_test.go). 

However I chose to use `hex.EncodeToString(md5.Sum(input))` because it seemed like the simplest thing to do here. 

> I guess hex.EncodeToString(hash[:]) can be simplified to hex.EncodeToString(hash), right?

The latter doesn't compile, because `hash` is type `[16]byte` but `hash[:]` is type `[]byte`.